lara-typescript/src/interactive-api-client/api.spec.ts

```diff
@@ -373,6 +391,111 @@ describe("api", () => {
       ]
     });
   });
+
+  it("does not yet implement getLibraryInteractiveList", () => {
+    expect(() => api.getLibraryInteractiveList({} as any)).toThrow();
+  });
+
+  describe("attachments support", () => {
+    const globalFetch = global.fetch;
+    const fetchMock = jest.fn();
+
+    beforeEach(() => {
+      global.fetch = fetchMock;
+      fetchMock.mockReset();
+    });
+
+    afterEach(() => {
+      global.fetch = globalFetch;
+    });
+
+    interface ITestAttachmentResponse {
+      requestContent: any[];
+      responseContent: any[];
+      resolvesTo?: any[];
+      rejectsWith?: any[];
+    }
+    const testWriteAttachmentResponse = (others: ITestAttachmentResponse) =>
+      testRequestResponse({
+        method: api.writeAttachment,
+        requestType: "getAttachmentUrl",
+        responseType: "attachmentUrl",
+        ...others
+      });
+
+    it("can write attachments", async () => {
+      const request: Partial<IWriteAttachmentRequest> = { name: "name.ext", content: "foo" };
+      const url = "https://concord.org/foo";
+      const apiResponse = { url };
+      fetchMock.mockReturnValue(Promise.resolve());
+      await testWriteAttachmentResponse({
+        requestContent: [request],
+        responseContent: [apiResponse],
+        resolvesTo: [undefined]
+      });
+      expect(fetchMock.mock.calls[0][0]).toBe(url);
+      expect(fetchMock.mock.calls[0][1]).toEqual({ method: "PUT", body: "foo" });
+    });
+
+    it("returns error when write attachment fails", async () => {
+      const request: Partial<IWriteAttachmentRequest> = { name: "name.ext", content: "foo" };
+      const response = { url: "https://concord.org/foo", fields: { foo: "bar" }};
+      fetchMock.mockImplementation(() => { throw new Error("error"); });
+      await testWriteAttachmentResponse({
+        requestContent: [request],
+        responseContent: [response],
+        rejectsWith: [new Error("error")]
+      });
+      expect(fetchMock.mock.calls[0][0]).toBe(response.url);
+      expect(fetchMock.mock.calls[0][1]).toEqual({ method: "PUT", body: "foo" });
+    });
+
+    const testReadAttachmentResponse = (others: ITestAttachmentResponse) =>
+      testRequestResponse({
+        method: api.readAttachment,
+        requestType: "getAttachmentUrl",
+        responseType: "attachmentUrl",
+        ...others
+      });
+
+    it("can read attachments", async () => {
+      const url = "https://concord.org/foo";
+      const apiResponse = { url };
+      const fetchResponse = { ok: true, text: () => "foo" };
+      fetchMock.mockReturnValue(Promise.resolve(fetchResponse));
+      await testReadAttachmentResponse({
+        requestContent: [{ name: "name.ext" }],
+        responseContent: [apiResponse],
+        resolvesTo: [fetchResponse]
+      });
+      expect(fetchMock.mock.calls[0][0]).toBe(url);
```

Comment:
Maybe you want to verify the method was a GET here.

Reply:
We're not passing a `method` explicitly here, but I've added a test that `fetch` was called with exactly one parameter, i.e. only a url and no additional options.